src/Compilers/Test/Core/SourceGeneration/TestGenerators.cs

```diff
@@ -110,6 +110,18 @@ public PipelineCallbackGenerator(Action<IncrementalGeneratorPipelineContext> reg
         public void Initialize(IncrementalGeneratorInitializationContext context) => context.RegisterExecutionPipeline(_registerPipelineCallback);
     }
 
+    internal sealed class PipelineCallbackGenerator2 : IIncrementalGenerator
```

Comment:
>PipelineCallbackGenerator2 [](http://example.com/codeflow?start=26&length=26)

This looks identical to `PipelineCallbackGenerator`. If so, why are two distinct classes needed?

Reply:
We have a long standing weirdness that you can only have one type of generator per driver. This is because of the way they're actually loaded in the compiler (as types from an assembly). We have an assert that checks there is only one type per driver in the ctor. 

Its on my backlog to audit the code and make sure we're not actually relying on that invariant and remove the assert and other `2` classes in that file.